cobc/tree.c

```diff
@@ -2769,7 +2769,7 @@ cb_build_numsize_literal (const void *data, const size_t size, const int sign)
 }
 
 cb_tree
-cb_build_alphanumeric_literal (const void *data, const size_t size)
```

Comment:
place the new `cb_build_literal_by_category` here instead of those two functions and use it accordingly